### ISSUE TITLE
CalibratedClassifierCV returns wrong predictions when used in pipeline

### ISSUE THREAD

**Issue by PatrickSager** [onnx/sklearn-onnx#301]
When using the CalibratedClassifierCV in a Pipeline that is exported to ONNX, then running the ONNX session returns completely different results than running the pipeline directly.

To reproduce the issue, take the tutorial example "Convert a pipeline with ColumnTransformer" (http://onnx.ai/sklearn-onnx/auto_examples/plot_complex_pipeline.html) and replace

clf = Pipeline(steps=[('preprocessor', preprocessor),
                      ('classifier', LogisticRegression(solver='lbfgs'))])

with 

clf = Pipeline(steps=[('preprocessor', preprocessor),
                      ('calibrator', CalibratedClassifierCV(base_estimator=LogisticRegression(solver='lbfgs', random_state=42), method='sigmoid', cv='warn'))])  

Then predict the same row once using the classifier and once using the ONNX model. The predicted probabilities are completely different.

(Attached is also the complete code).
[titanic.zip](https://github.com/onnx/sklearn-onnx/files/3784470/titanic.zip)

**Comment by PatrickSager** [comment]
After further investigation, it turned out that the CalibratedClassifierCV returns also wrong results if it is used directly (not in a pipeline).

Here's the code that shows the difference between the predictions when calling the trained classifier directly and when calling the loaded ONNX model:

`from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.calibration import CalibratedClassifierCV
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType
import onnxruntime as rt
import numpy

# Load data
iris = load_iris()
X, y = iris.data, iris.target
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

# Fit model
#clr = LogisticRegression()
clr = CalibratedClassifierCV(base_estimator=LogisticRegression(), method='sigmoid')

clr.fit(X_train, y_train)

# Predict with the fitted model
pred_skl = clr.predict(X_test)
print("Predictions from skl:", pred_skl)

# Export to ONNX
initial_type = [('float_input', FloatTensorType([-1, X_train.shape[1]]))]
onx = convert_sklearn(clr, initial_types=initial_type)
with open("exported_model.onnx", "wb") as f:
    f.write(onx.SerializeToString())

# Predict with exported ONNX model
sess = rt.InferenceSession("exported_model.onnx")
input_name = sess.get_inputs()[0].name
label_name = sess.get_outputs()[0].name
pred_onx = sess.run([label_name], {input_name: X_test.astype(numpy.float32)})[0]
print("Predictions from ONNX:", pred_onx)`

Returned results:
`Predictions from skl: [1 0 2 1 1 0 1 2 1 1 2 0 0 0 0 1 2 1 1 2 0 2 0 2 2 2 2 2 0 0 0 0 1 0 0 2 1 0]
Predictions from ONNX: [1 0 1 1 1 0 1 1 1 1 1 0 0 0 0 1 1 1 1 1 0 1 0 1 1 1 1 2 0 0 0 0 1 0 0 1 1 0]`

(Besides, changing the CalibratedClassifierCV in the code above to LogisticRegression, the two outputs match).

**Comment by prabhat00155** [comment]
Oh, I didn't mention that for you to tinker with the file itself. I just linked it to point out the difference. Since we only implement predict_proba() and not decision_function(), we don't support CalibratedClassifierCV with predictors with decision_function(). That's something we could look at implementing, but it's going to be a major change and would require some time. Is there any particular reason you need to go with LogisticRegression? You can pick a predictor which doesn't have decision_function().

**Comment by prabhat00155** [comment]
Yes, you may use these for the time being. We have started planning on adding decision_function() in the converters, and I'll prioritise this work. I'll be able to give an estimated timeline soon.

**Comment by PatrickSager** [comment]
Perfect. Looking forward to hear from you.
